recipe/common.php

```diff
@@ -46,41 +47,83 @@
  * Configuration
  */
 
-set('keep_releases', 5);
-
-set('repository', ''); // Repository to deploy.
-
-set('shared_dirs', []);
-set('shared_files', []);
-
-set('copy_dirs', []);
-
-set('writable_dirs', []);
-set('writable_mode', 'acl'); // chmod, chown, chgrp or acl.
-set('writable_use_sudo', false); // Using sudo in writable commands?
-set('writable_recursive', true); // Common for all modes
-set('writable_chmod_mode', '0755'); // For chmod mode
-set('writable_chmod_recursive', true); // For chmod mode only (if is boolean, it has priority over `writable_recursive`)
-
-set('http_user', false);
-set('http_group', false);
+set('keep_releases', 5, function ($value) {
+    if (is_integer($value) && $value >= -1) {
+        return true;
+    }
+    throw new ConfigurationException("Config error: \"keep_releases\" value must be an integer equal or bigger than \"-1\".");
+});
 
-set('clear_paths', []);         // Relative path from release_path
-set('clear_use_sudo', false);    // Using sudo in clean commands?
+set('repository', '', function ($value) {
+    return is_string($value); 
+}); // Repository to deploy.
+
+set('shared_dirs', [], 'is_array');
+set('shared_files', [], 'is_array');
+set('copy_dirs', [], 'is_array');
+set('writable_dirs', [], 'is_array');
+set('writable_mode', 'acl', function ($value) {
+    return $value === 'chmod'
+        || $value === 'chown'
+        || $value === 'chgrp'
+        || $value === 'acl'; 
+}); // chmod, chown, chgrp or acl.
+
+set('writable_use_sudo', false, 'is_bool'); // Using sudo in writable commands?
+set('writable_recursive', true, 'is_bool'); // Common for all modes
+
+set('writable_chmod_mode', '0755', function ($value) {
+    $chmodPattern = '/^[ugoa]*([-+=]([rwxXst]*|[ugo]))+|[-+=]?[0-7]+$/';
+    return preg_match($chmodPattern, $value);
+}); // For chmod mode
+
+set('writable_chmod_recursive', true, 'is_bool'); // For chmod mode only (if is boolean, it has priority over `writable_recursive`)
+set('http_user', false, 'is_bool');
+set('http_group', false, 'is_bool');
+
+set('clear_paths', [], function ($value) {
+    if (!is_array($value)) {
+        return false;
+    }
+    foreach ($value as $dir) {
```

Comment:
`clear_paths` works the same as other dirs. Not able to use `is_dir`.

Reply:
Fixing it as well.